### PR TITLE
FIX: pass nonposx/y args through loglog etc

### PR DESCRIPTION
**Pull request by jklymak**
## PR Summary

This is a quick fix for #10007 that simply passes the `nonposx` `nonposy` args through to `set_xscale` etc.  

happy for someone who knows everything that happened in here to do this properly and close this...

Added a pair of tests (and two new images, I'm afraid).

## PR Checklist

- [ ] Has Pytest style unit tests
- [ ] Code is PEP 8 compliant
- [ ] New features are documented, with examples if plot related
- [ ] Documentation is sphinx and numpydoc compliant
- [ ] Added an entry to doc/users/next_whats_new/ if major new feature (follow instructions in README.rst there)
- [ ] Documented in doc/api/api_changes.rst if API changed in a backward-incompatible way

### CODE REVIEW

**Comment by anntzer** [comment]
This should work, but I would prefer something like
```
diff --git a/lib/matplotlib/axes/_axes.py b/lib/matplotlib/axes/_axes.py
index 27efb72b6..f0dc82663 100644
--- a/lib/matplotlib/axes/_axes.py
+++ b/lib/matplotlib/axes/_axes.py
@@ -1560,11 +1560,9 @@ class Axes(_AxesBase):
         """
         if not self._hold:
             self.cla()
-        d = {'basex': kwargs.pop('basex', 10),
-             'subsx': kwargs.pop('subsx', None),
-             }
-
-        self.set_xscale('log', **d)
+        self.set_xscale(
+            'log', **{k: kwargs.pop(k) for k in ['basex', 'subsx', 'nonposx']
+                      if k in kwargs})
         b = self._hold
         self._hold = True  # we've already processed the hold
         l = self.plot(*args, **kwargs)
```
which avoids having to hard-code the defaults into _axes.py as well (just use the defaults which are already correctly set in scale.py).

**Comment by jklymak** [comment]
Done, though couldn't bring myself to put it all on one line ;-)

File: lib/matplotlib/tests/test_axes.py [hunk 1]
```diff
@@ -4964,6 +4964,26 @@ def test_loglog():
     ax.tick_params(length=15, width=2, which='minor')
 
 
+@image_comparison(baseline_images=["test_loglog_nonposx_clip"],
```
Comment (anntzer) [inline lib/matplotlib/tests/test_axes.py]: save an image by using subplots? :-)

Comment (jklymak) [reply lib/matplotlib/tests/test_axes.py]: Umm yeah that’s fair.  I can actually just plot twice on the same axis.  Or four times to get all four cases.   Stand by

File: lib/matplotlib/tests/test_axes.py [hunk 2]
```diff
@@ -4964,24 +4964,23 @@ def test_loglog():
...
     x[4] = -10
-    ax.loglog(x, y**3, lw=5, nonposx='mask')
+    for nn, mcx in enumerate(['mask', 'clip', '']):
+        for mm, mcy in enumerate(['mask', 'clip', '']):
+            kws = {}
+            if mcx:
+                kws['nonposx'] = mcx
+            if mcy:
+                kws['nonposy'] = mcy
+            print(kws)
```
Comment (anntzer) [inline lib/matplotlib/tests/test_axes.py]: no need to print?

Comment (jklymak) [reply lib/matplotlib/tests/test_axes.py]: gah

**Comment by anntzer** [comment]
Squash the old images away from the history?  Otherwise lgtm.

**Comment by jklymak** [comment]
Squashed.  

BTW the nightly builds are failing for some reason (polar related?)

**Pull request merged**